pkg/utils/test/testf/testf_witht_test.go

```diff
@@ -0,0 +1,239 @@
+package testf_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rs/xid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/fakeclient"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/matchers/jq"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/testf"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGet(t *testing.T) {
+	g := NewWithT(t)
+
+	cm := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: gvk.ConfigMap.GroupVersion().String(),
+			Kind:       gvk.ConfigMap.Kind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      xid.New().String(),
+		},
+	}
+
+	cl, err := fakeclient.New(&cm)
+	g.Expect(err).ShouldNot(HaveOccurred())
+	g.Expect(cl).ShouldNot(BeNil())
+
+	tc, err := testf.NewTestContext(testf.WithClient(cl))
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	key := client.ObjectKeyFromObject(&cm)
+
+	matcher := And(
+		jq.Match(`.metadata.namespace == "%s"`, cm.Namespace),
+		jq.Match(`.metadata.name == "%s"`, cm.Name),
+	)
+
+	t.Run("Get", func(t *testing.T) {
+		wt := tc.NewWithT(t)
+
+		v, err := wt.Get(gvk.ConfigMap, key).Get()
+
+		g.Expect(err).ShouldNot(HaveOccurred())
+		g.Expect(v).Should(matcher)
```

Comment:
May be the matcher name(s) could be more readable, like `Should(MatchMetadata)` ?

Reply:
done

Reply:
> done

How about other (than Get) tests?

Reply:
done